transmogrifier/sources/xml/marc.py

```diff
@@ -972,6 +954,47 @@ def get_publication_frequency(cls, source_record: Tag) -> list[str] | None:
             )
         ] or None
 
+    @classmethod
+    def get_publishers(cls, source_record: Tag) -> list[timdex.Publisher] | None:
+        publishers = []
+        for publisher_marc_field in ["260", "264"]:
+            publishers.extend(
+                [
+                    timdex.Publisher(
+                        name=publisher_name.rstrip(".,") if publisher_name else None,
+                        date=publisher_date.rstrip(".,") if publisher_date else None,
+                        location=(
+                            publisher_location.rstrip(" :")
+                            if publisher_location
+                            else None
+                        ),
+                    )
+                    for datafield in source_record.find_all(
+                        "datafield", tag=publisher_marc_field
+                    )
+                    if any(
+                        [
+                            (
+                                publisher_name := cls.get_single_subfield_string(
+                                    datafield, "b"
+                                )
+                            ),
+                            (
+                                publisher_date := cls.get_single_subfield_string(
+                                    datafield, "c"
+                                )
+                            ),
+                            (
+                                publisher_location := cls.get_single_subfield_string(
+                                    datafield, "a"
+                                )
+                            ),
+                        ]
+                    )
+                ]
+            )
+        return publishers or None
```

Comment:
To me, it feels like readability is being compromised here even if this is more efficient but curious what others think

Reply:
This has been [addressed in our comment thread](https://github.com/MITLibraries/transmogrifier/pull/203#discussion_r1697118093) above. :)